### ISSUE TITLE
Search for Map CRS coordinates

### ISSUE THREAD

**Issue by tdipisa** [geosolutions-it/MapStore2#10136]
## Description

It is requested with this issue to improve the CoordinatesSearch tool to be able to use coordinates expressed in the selected Map CRS, now it is possible a search for coordinates in the defined on the map, now it's only possible by Lat/Long.

![Image](https://github.com/geosolutions-it/MapStore2/assets/1280027/de6819a1-abb2-4491-b403-608747255d90)

A proper entry in the current coordinate type selector (decimal/aeronautical) should allow to select the current Map CRS to setup the tool accordingly (e.g. use current Map CRS)

![Image](https://github.com/geosolutions-it/MapStore2/assets/1280027/9c0dea9c-3770-43b4-ab41-407589fdd45f)

Thank into account that the map CRS can vary according to the CRS selected din the [CRS Selector](https://docs.mapstore.geosolutionsgroup.com/en/latest/user-guide/footer/#crs-selector).

**What kind of improvement you want to add?** (check one with "x", remove the others)
- [X] Minor changes to existing features
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## Design

It would be good to implement the following UI/UX:

- As already done for "Search by location name" in #9751, include a sub menu item for "Search by coordinates"
- Selecting "Search by coordinates" the tool maintain the current behavior with the usual decimal and aeronautical options

![image](https://github.com/geosolutions-it/MapStore2/assets/1280027/cf666e88-9bdb-4121-b3bd-7f3a30159ebc)

- Selecting "Current map CRS" the possibility to switch between decimal/aeronautical disappear from the UI and the tool setup change by switching to map CRS coordinates

![image](https://github.com/geosolutions-it/MapStore2/assets/1280027/59993cf7-7190-4abb-9cf1-7c4e05d44223)

- The "Current map CRS" is available only if the the current Map CRS is not the one normally used by decimal and aeronautical coordinates, which means EPSG:4326

![image](https://github.com/geosolutions-it/MapStore2/assets/1280027/d72ecb0b-6eaa-4074-976c-35052adf08f8)

- As coordinate fields validator when "Current map CRS" is selected, the current one should be preserved which means:
   - The entered value is a number
   - The entered coordinates are within the CRS bounds
   - Same decimal separators should be used according locale as is also for other plugins where geographic coordinates are exposed 

- Current functionalities have to be maintained also when "Current map CRS" is selected e.g. when search is performed, the tool has to add the marker to the map and center / zoom the point as usual

## Other useful information
The functionality is actually implemented here:

https://github.com/geosolutions-it/MapStore2/blob/master/web/client/components/mapcontrols/searchcoordinates/CoordinatesSearch.js

`zoomAndAddPoint` already send the  CRS. This action is actually intercepted by `epics/search.js` --> `zoomAndAddPointEpic`

https://github.com/geosolutions-it/MapStore2/blob/2556203c9ebeb77a5157d1718a9d27b15c98fdbb/web/client/epics/search.js#L273-L297

The epic does 2 operation.
- Add a layer with the marker. **This is in 4326, and should support also the map CRS (or be converted).**
` zoomToPoint` should already support a different CRS

**Comment by dsuren1** [comment]
@offtherailz 
For CRS `x,y` input fields, should we validate the limits entered like we do with decimal or simply check if the value is a number?

**Comment by tdipisa** [comment]
**Screenshot**

- Upon changing the CRS, the x,y value is reprojected and updated value is reflected in the fields
- Upon changing to decimal/aeronautical fields, the lat,lng is reprojected to 4326 and the updated value is reflected in the fields

<img width="1036" alt="320765887-d381b0a5-cc56-4c4b-8f94-f1be4e30042e" src="https://github.com/geosolutions-it/MapStore2/assets/1280027/feb9dfeb-aa39-4f1d-817f-a9cefec5df7b">
<img width="1036" alt="320765887-d381b0a5-cc56-4c4b-8f94-f1be4e30042e (1)" src="https://github.com/geosolutions-it/MapStore2/assets/1280027/d1b08bd8-e4fc-4e23-9586-9e1c31358d14">

@offtherailz
For CRS x,y input fields, what should be the validations on these fields, only check if the value is a number?

**Comment by offtherailz** [comment]
About the validation, we can maybe verify that the coordinates are in the defined projection range. 
For the entries of the menu, it is a little confusing that we have 2 entries for 4326 format and one for the map context.
It could be more logical to be able to select the projection of the coordinates in a separate tool, and when they are 4326, the 2 formats are allowed.
Maybe a "SELECT PROJECTION... " entry in the menu that allows you to select a projection, with the current projection selected and the map projection highlighted could be clear enough.
It requires 1 click more, but is less specific and more clear.

**Comment by tdipisa** [comment]
> About the validation, we can maybe verify that the coordinates are in the defined projection range. For the entries of the menu, it is a little confusing that we have 2 entries for 4326 format and one for the map context. It could be more logical to be able to select the projection of the coordinates in a separate tool, and when they are 4326, the 2 formats are allowed. Maybe a "SELECT PROJECTION... " entry in the menu that allows you to select a projection, with the current projection selected and the map projection highlighted could be clear enough. It requires 1 click more, but is less specific and more clear.

If so let's discuss a bit more on the design part before proceeding with this issue @dsuren1

**Comment by tdipisa** [comment]
I've updated the issue description clarifying the UI/UX part. Issue unblocked now and assigned to @mahmoudadel54
